2020-04-introducing-mdjs-live-demos-everywhere/README.md

```diff
@@ -0,0 +1,360 @@
+Users should at least on the high level understand what the are using for what and why.
+
+On the web, we have many many different ways of writing documentation.
+However, one thing almost all of them have in common is that they rely on [Markdown](https://en.wikipedia.org/wiki/Markdown) or of variations of it.
+
+It is supported literally everywhere (vscode, atom, github, gitlab, dev.to, npmjs, ...)
+
+## For tools that do not run in the browser
+
+In that case, you will mostly share code snippets that people will need to run in their own projects so traditional static site generators like [Docusaurus](https://docusaurus.io/), [VuePress](https://vuepress.vuejs.org/), [Gatsby](https://www.gatsbyjs.org/), ... work great. All of them have full support for Markdown and therefore can create beautiful documentation pages with code snippets/highlighting, ...
+
+Honestly, if that is your use case almost everything should be possible as long as you feel comfortable with the tools ecosystem/framework.
+
+## For (visual) components that do run in the browser
+
+Here people do expect a live demo to see different options in action. So pure Markdown is usually not enough as we now want to actually execute code and "insert" our working component. This requires specialized handling for each framework.
+
+### Vue
+
+For Vue you can use for example VuePress which auto registers all vue components in a certain folder and then you can use as normal html tags as Markdown allows html
+
+```
+.
+└─ .vuepress
+  └─ components
+      ├─ demo-1.vue
+```
+
+```html
+<demo-1 />
+```
+
+- supports vue components and has "magical" import for them
+- no support for generic javascript or passing properties to components
+
+### React
+
+For React you can use [mdx](https://mdxjs.com/) which extends Markdown with JSX support. Mdx is available via multiple tools like [Gatsby](https://www.gatsbyjs.org/), [docz](https://www.docz.site/), [storybook](https://storybook.js.org/), ...
+
+```md
+import { Chart } from '../components/chart'
+
+# Here’s a chart
+
+The chart is rendered inside our MDX document.
+<Chart />
+```
+
+- supports import/export JavaScript
+- passes everything through JSX
+- looks "bad" on github, requires special tools in editors to get highlighting
+
+## Limitations
+
+What all these specialized tools have in common is that they require a specific build tooling setup to work.
+For web components, none of that is actually needed. Markdown already allows for HTML. The only missing piece is how to load a web component?
+
+## Introducing Markdown with JavaScript (mdjs)
+
+The primary goals are
+
+- minimal complexity
+- follows progressive enhancement
+- markdown needs to still make sense
+- code highlighting in editors without additional tools
+- looks good on github/gitlab/any source code management tool
+
+The fundamental idea is laughable simple. We "enhance" a code fence block with additional meta data `js script`.
```

Comment:
```suggestion
The fundamental idea is laughably simple. We "enhance" a code fence block with additional meta data `js script`.
```

I know this is not your intention, but I feel like "laughably simple" comes across a little 'arrogant', maybe theres a better way you can word this?

Reply:
How about
```suggestion
The fundamental idea is extremely simple. We "enhance" a code fence block with additional meta data `js script`.
```

or

```suggestion
The fundamental idea is seems almost too simple to be true. We "enhance" a code fence block with additional meta data `js script`.
```